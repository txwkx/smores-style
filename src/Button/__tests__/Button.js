import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Button } from '../Button'

test('snapshot', () => {
  const { container } = render(<Button />)

  expect(container.firstChild).toMatchSnapshot()
})

test('renders Green button', () => {
  const { getByText } = render(
    <Button color="green" data-testid="testBtn">
      Push me
    </Button>,
  )
  const button = getByText(/push me/i)

  expect(button).toHaveTextContent('Push me')
  expect(button).toHaveAttribute('color', 'green')
})

test('renders Disabled Blue button', () => {
  const { container } = render(
    <Button disabled color="blue">
      Blue
    </Button>,
  )

  expect(container.firstChild).toHaveAttribute('color', 'blue')
  expect(container.firstChild).toBeDisabled()
})

test('render Block button with 100% width', () => {
  const { container } = render(<Button block>Block</Button>)

  expect(container.firstChild).toHaveStyleRule('width', '100%')
})

test('cant click Disabled button', () => {
  const { getByText } = render(<Button disabled>Click</Button>)
  const button = getByText(/click/i)

  // debug()
  fireEvent.click(button)
  expect(button).toBeDisabled()
})
