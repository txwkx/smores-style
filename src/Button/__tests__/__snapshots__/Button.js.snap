// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  border: none;
  border-radius: 8px;
  font-size: 14px;
  padding: 18px 16px 14px;
  outline: none;
  cursor: pointer;
  width: auto;
  background-color: #545E7E;
  color: #ffffff;
}

.c0:hover:not([disabled]) {
  background-color: #363F59;
}

.c0:active:not([disabled]) {
  background-color: #112035;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

@media (min-width:768px) {
  .c0 {
    padding: 19px 16px 15px;
    font-size: 16px;
  }
}

<button
  class="c0"
  color="blue"
/>
`;
